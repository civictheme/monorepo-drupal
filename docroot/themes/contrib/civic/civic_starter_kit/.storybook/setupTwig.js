const twigDrupal = require('twig-drupal-filters');

/**
 * Configures and extends a standard twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
