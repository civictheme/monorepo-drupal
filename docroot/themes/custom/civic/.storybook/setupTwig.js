const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');

module.exports.namespaces = {
  atoms: resolve(__dirname, '../', 'components/01-atoms'),
  molecules: resolve(__dirname, '../', 'components/02-molecules'),
  organisms: resolve(__dirname, '../', 'components/03-organisms'),
  templates: resolve(__dirname, '../', 'components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
