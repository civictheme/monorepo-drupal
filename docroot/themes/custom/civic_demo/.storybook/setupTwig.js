const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');


module.exports.namespaces = {
  atoms: resolve(__dirname, '../', '.storybook-components/01-atoms'),
  molecules: resolve(__dirname, '../', '.storybook-components/02-molecules'),
  organisms: resolve(__dirname, '../', '.storybook-components/03-organisms'),
  templates: resolve(__dirname, '../', '.storybook-components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
