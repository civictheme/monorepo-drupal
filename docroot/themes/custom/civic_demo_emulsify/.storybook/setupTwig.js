const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');
const twigBEM = require('bem-twig-extension');
const twigAddAttributes = require('add-attributes-twig-extension');

module.exports.namespaces = {
  atoms: [
    resolve(__dirname, '../../civic_emulsify', 'components/01-atoms'),
    resolve(__dirname, '../', 'components/01-atoms'),
  ],
  molecules: [
    resolve(__dirname, '../../civic_emulsify', 'components/02-molecules'),
    resolve(__dirname, '../', 'components/02-molecules'),
  ],
  organisms: [
    resolve(__dirname, '../../civic_emulsify', 'components/03-organisms'),
    resolve(__dirname, '../', 'components/03-organisms'),
  ],
  templates: [
    resolve(__dirname, '../../civic_emulsify', 'components/04-templates'),
    resolve(__dirname, '../', 'components/04-templates'),
  ]
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigBEM(twig);
  twigDrupal(twig);
  twigAddAttributes(twig);
  return twig;
};
