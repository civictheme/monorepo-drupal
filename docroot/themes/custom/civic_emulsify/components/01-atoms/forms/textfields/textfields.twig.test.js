import { join } from 'path';
import { render, Twig } from 'twig-testing-library';

import { setupTwig } from '../../../../.storybook/setupTwig';

setupTwig(Twig);

describe('textfields', () => {
  it('can render textfield form items', async () => {
    const { container } = await render(join(__dirname, 'textfields.twig'));

    expect(container).toMatchSnapshot();
  });
});
