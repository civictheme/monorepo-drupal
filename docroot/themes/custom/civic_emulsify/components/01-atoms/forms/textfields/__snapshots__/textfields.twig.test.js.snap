// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textfields can render textfield form items 1`] = `
<div>
  <div
    class="form-item"
  >
    
  
    <label
      class="form-item__label form-item__label--required form-item__label--textfield"
      for="edit-name"
    >
      Name
    </label>
    
  
    <input
      aria-required="true"
      autocapitalize="none"
      autocorrect="none"
      class="form-item__textfield form-item__textfield--required"
      id="edit-name"
      maxlength="60"
      name="name"
      placeholder="First, Middle, Last"
      required="required"
      size="60"
      spellcheck="false"
      type="text"
      value=""
    />
    

  </div>
  


  <div
    class="form-item"
  >
    
  
    <label
      class="form-item__label form-item__label--required form-item__label--textfield"
      for="edit-pass"
    >
      Password
    </label>
    
  
    <input
      aria-required="true"
      class="form-item__textfield form-item__textfield--password form-item__textfield--required"
      id="edit-pass"
      maxlength="128"
      name="pass"
      required="required"
      size="60"
      type="password"
    />
    
  
    <div
      class="form-item__description"
    >
      
    Enter the password that accompanies your username.
  
    </div>
    

  </div>
  


  <form>
    
  
    <fieldset
      class="form-fieldset"
    >
      
  
      <legend
        class="heading-2 with-icon"
      >
        Web and Email Address (fieldset example)
      </legend>
      
    
      <div
        class="form-item"
      >
        
       
        <label
          class="form-item__label form-item__label--textfield"
          for="webaddress"
        >
          Web Address
        </label>
        
       
        <input
          class="form-item__textfield"
          id="webaddress"
          placeholder="http://yoursite.com"
          type="url"
        />
        
     
      </div>
      

     
      <div
        class="form-item"
      >
        
       
        <label
          class="form-item__label form-item__label--textfield"
          for="emailaddress"
        >
          Email Address
        </label>
        
       
        <input
          class="form-item__textfield"
          id="emailaddress"
          placeholder="name@email.com"
          type="email"
        />
        
     
      </div>
      
  
    </fieldset>
    
 
  </form>
  

 
  <div
    class="form-item"
  >
    
   
    <label
      class="form-item__label form-item__label--textfield"
      for="text"
    >
      Number Input
    </label>
    
   
    <input
      class="form-item__textfield"
      id="text"
      pattern="[0-9]*"
      placeholder="Enter a Number"
      type="number"
    />
    
 
  </div>
  

 
  <div
    class="form-item"
  >
    
   
    <label
      class="form-item__label form-item__label--textfield"
      for="textarea"
    >
      Textarea
    </label>
    
   
    <textarea
      class="form-item__textfield form-item__textarea"
      cols="48"
      id="textarea"
      placeholder="Enter your message here"
      rows="8"
    />
    
 
  </div>
  

</div>
`;
